### ISSUE TITLE
`<li>` marker is invisible on otherwise empty page

### ISSUE THREAD

**Issue by Psychpsyo** [LadybirdBrowser/ladybird#2433]
### Summary

When I open `data:text/html,<li>` in Ladybird, the page looks completely empty.

~~This might be related to #2432~~
EDIT: This is to spec, but inconsistent with Chrome and also with quirks-mode Firefox.

### Operating system

Linux

### Steps to reproduce

1. Open `data:text/html,<li>`.

### Expected behavior

There should be a single black dot on the page.

### Actual behavior

There is nothing on the page.

### URL for a reduced test case

`data:text/html,<li>`

### HTML/SVG/etc. source for a reduced test case

```html
<li>
```

### Log output and (if possible) backtrace

```shell
Nothing
```

### Screenshots or screen recordings

![image](https://github.com/user-attachments/assets/8d2a0b4a-c29a-4a19-8c61-6b24085c969a)

### Contribute a patch?

- [ ] I’ll contribute a patch for this myself.

**Comment by Psychpsyo** [comment]
> I'm not sure if this is a bug or not, it's inconsistent between browsers, and isn't specced as far as I can tell.

Just checked the specs and, as far as I can tell, this is specced:
[15.3.7 Lists](https://html.spec.whatwg.org/multipage/rendering.html#lists) in the HTML spec says that &lt;li&gt;-elements should have `display: list-item;` by default.
[2. Box Layout Modes: the 'display' property](https://drafts.csswg.org/css-display-4/#the-display-properties) in css-display-4 says that that expands to `block flow list-item`, meaning the outer display type is `block`, making the element a `block container`.
[3.5. Positioning Markers: The 'list-style-position' property](https://drafts.csswg.org/css-lists-3/#list-style-position-property) in css-lists-3 says that the initial value for `list-style-position` is `outside`, which, for block containers means that the marker box is placed outside the element's principal box, on the inline-start side.

TL;DR: The marker should be outside of the &lt;li&gt;.

Also, Firefox's quirks stylesheet is the way it is ["for backward compatibility"](https://bugzilla.mozilla.org/show_bug.cgi?id=1049#c4), whatever that means.
Chrome behaves the way it does because its parser does some magic [to match what internet explorer did back then](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/style/computed_style.cc;l=2676-2679).

Adding a sufficient margin on the body does reveal the marker in Ladybird, so yes, this is actually to spec and not a bug.
Not sure if we should bother with matching the other browsers on this, especially since Chromium also wants to get rid of their magic and any document that can trigger this is non-conforming anyways.

**Comment by Psychpsyo** [comment]
One more bit I just stumbled upon:

Microsoft has actually documented their non-standard quirk for this here:
https://learn.microsoft.com/en-us/openspecs/ie_standards/ms-css21/7b7dd76a-ed48-439e-b7b3-939100cb403d

> # 7.2 The effects of quirks mode emulation
> - **List item markers.** The default value of **list-style-position** is "inside" if the list-item is outside a &lt;UL&gt; or &lt;OL&gt; element, and "outside" otherwise.

(I'm not saying we should support this, just that I found the source of the nonsense)

**Comment by Psychpsyo** [comment]
This has been fixed by the PR above.
